### PR TITLE
Fixes #411 bean-validator extension

### LINKED ISSUE #411: bean-validator extension

### PR DESCRIPTION
**Pull request by davsclaus**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus as follow up of this PR we should check if we can leverage anything from https://github.com/quarkusio/quarkus/tree/master/extensions/hibernate-validator

**Comment by lburgazzoli** [comment]
ok to test

**Comment by davsclaus** [comment]
> @davsclaus as follow up of this PR we should check if we can leverage anything from https://github.com/quarkusio/quarkus/tree/master/extensions/hibernate-validator

Okay I am now using the hibernate-validator extension. See last commit

**Comment by lburgazzoli** [comment]
ok to test

File: poms/bom-deployment/pom.xml [hunk 1]
```diff
@@ -303,6 +308,11 @@
                 <artifactId>camel-quarkus-xslt-deployment</artifactId>
                 <version>${camel-quarkus.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.camel.quarkus</groupId>
+                <artifactId>camel-quarkus-bean-validator-deployment</artifactId>
```
Comment (ppalaga) [inline poms/bom-deployment/pom.xml]: camel-quarkus-bean-validator-deployment seems to be twice here.

**Comment by lburgazzoli** [comment]
ok to test

**Comment by davsclaus** [comment]
ok to test

**Comment by davsclaus** [comment]
So it looks like hibernate-validator has a constract that may use jodatime

Error: Class initialization of org.hibernate.validator.internal.constraintvalidators.bv.time.pastorpresent.PastOrPresentValidatorForReadablePartial failed. Use the option --initialize-at-run-time=org.hibernate.validator.internal.constraintvalidators.bv.time.pastorpresent.PastOrPresentValidatorForReadablePartial to explicitly request delayed initialization of this class.
Original exception that caused the problem: java.lang.NoClassDefFoundError: org/joda/time/ReadableInstant

**Comment by davsclaus** [comment]
Adding joda-time 1.4 makes the native image build, but at test/runtime you get this error

Caused by: javax.validation.NoProviderFoundException: Unable to create a Configuration, because no Jakarta Bean Validation provider could be found. Add a provider like Hibernate Validator (RI) to your classpath.
	at javax.validation.Validation$GenericBootstrapImpl.configure(Validation.java:291)
	at org.apache.camel.component.bean.validator.ValidatorFactories.buildValidatorFactory(ValidatorFactories.java:49)
	at org.apache.camel.component.bean.validator.BeanValidatorEndpoint.createProducer(BeanValidatorEndpoint.java:69)
	at org.apache.camel.support.DefaultEndpoint.createAsyncProducer(DefaultEndpoint.java:196)
	at org.apache.camel.impl.engine.ServicePool$SinglePool.acquire(ServicePool.java:234)
	at org.apache.camel.impl.engine.ServicePool$SinglePool.acquire(ServicePool.java:221)
	at org.apache.camel.impl.engine.ServicePool.acquire(ServicePool.java:112)
	at org.apache.camel.impl.engine.DefaultProducerCache.acquireProducer(DefaultProducerCache.java:113)
	... 38 more
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 60.13 s <<< FAILURE! - in org.apache.camel.quarkus.component.bean.validator.it.BeanValidatorIT
[ERROR] test  Time elapsed: 0.01 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [org.apache.camel.quarkus.component.bean.validator.it.BeanValidatorIT]: Unable to start native image in 60s
Caused by: java.lang.RuntimeException: Unable to start native image in 60s

**Comment by lburgazzoli** [comment]
Raised an issue on quarkus: https://github.com/quarkusio/quarkus/issues/5508

**Comment by jamesnetherton** [comment]
Wondering whether we want Camel to build its own `ValidatorFactory`? The Quarkus extension is [already doing this](https://github.com/quarkusio/quarkus/blob/master/extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java#L110).

**Comment by lburgazzoli** [comment]
yup, we need to do that

**Comment by lburgazzoli** [comment]
working on a workaround to see if it fixes this issue, if it works, we may then need to enhance a bit the component

**Comment by lburgazzoli** [comment]
mh no, it does not fix this issue

**Comment by lburgazzoli** [comment]
ok to test

**Pull request merged**